### PR TITLE
feat: added rebase interactive

### PR DESCRIPTION
**Pull request by kessejones**
adds the possibility to perform interactive rebase

### CODE REVIEW

**Comment by kessejones** [comment]
@sotte

**Comment by sotte** [comment]
I'll test it out during the next rebase. Thanks @kessejones.

File: lua/neogit/lib/rpc.lua [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  rpc:connect()
+
+  return rpc
+end
+
+function RPC:connect()
+  self.ch = fn.sockconnect("pipe", self.address, { rpc = true })
+end
+
+function RPC:desconnect()
```
Comment (TimUntersberger) [inline lua/neogit/lib/rpc.lua:34]: This should be disconnect instead of desconnect

**Review by TimUntersberger — COMMENTED** [review]
What is the neovim RPC for? Is it just for testing?

Also please fix the linting errors.

Thank you!

**Comment by kessejones** [comment]
About neovim RPC

It is being used to communicate with the neovim process started by running `git rebase -i`.
To detect the opening of the `git-rebase-todo` file, I used the environment variables `GIT_SEQUENCE_EDITOR` and `GIT_EDITOR` to send a custom neovim command, which connects to the parent process and opens the file in a buffer, at the end of editing this file we send a signal to the child process to terminate it.

**Comment by TimUntersberger** [comment]
Thanks for explaining! 😄

**Pull request merged**